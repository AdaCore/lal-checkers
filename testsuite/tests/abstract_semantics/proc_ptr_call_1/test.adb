procedure Test is
   X : Integer;

   procedure F is
   begin
      X := 12;
   end F;

   P : access procedure := F'Access;
begin
   P.all;
end Test;
