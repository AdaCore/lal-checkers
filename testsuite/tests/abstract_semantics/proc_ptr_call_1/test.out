{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "$stack": "({}, 0)", 
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[12, 12]'}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "assume0": [
      {
        "trace:": [
          "assign0",
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[all-path]}"
        }
      }
    ]
  },
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({1: '[12, 12]'}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ],
    "assume0": [
      {
        "trace:": [
          "assign0",
          "assume0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{Subprogram <SubpBody [\"F\"] 4:4-7:10> capturing (0x00000001)}"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "$stack": "({}, 0)", 
          "P": "{[all-path]}"
        }
      }
    ]
  }
}